### PR TITLE
Simplify PredictionPipeline

### PR DESCRIPTION
**Pull request by FynnBe**
builds ontop of #105

### CODE REVIEW

File: bioimageio/core/prediction_pipeline/_prediction_pipeline.py [hunk 1]
```diff
@@ -144,26 +91,6 @@ def input_specs(self):
     def output_specs(self):
         return self._output_specs
 
-    @property
```
Comment (k-dominik) [inline bioimageio/core/prediction_pipeline/_prediction_pipeline.py:147]: so, with these properties gone, are they supposed to be queried from the model?

**Review by k-dominik — APPROVED** [review]
cool to have all the NN dependency checking for tests in a single place now :)

File: bioimageio/core/prediction_pipeline/_prediction_pipeline.py (hunk 1 above)
Comment (FynnBe) [reply bioimageio/core/prediction_pipeline/_prediction_pipeline.py:147]: they are supposed to be accessed through `input_specs` and `output_specs` (which are set from `model.inputs` and `model.outputs` respectively. This avoids redefining everything as the (non-raw/derived) nodes are now part of this core module.

Comment (k-dominik) [reply bioimageio/core/prediction_pipeline/_prediction_pipeline.py:147]: good to know. Sounds reasonable :)

**Pull request merged**